Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLReader.java

```diff
@@ -0,0 +1,573 @@
+                }
+            } else if (child instanceof Comment) {
+                // Ignore XML comments
+                continue;
+            } else {
+                throw new ClassCastException(
+                        "Found unknown subclass of org.w3c.dom.Node: " + child.getClass().getName());
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Static method to get the elements from an XML stream. This is used in the
+     * constructors from file. The stream will be closed before the method
+     * exits.
+     *
+     * @param input
+     *            input to parse. Will be closed afterwards.
+     * @param encoding
+     *            character encoding to use
+     * @return the root element
+     * @throws SAXException
+     *             if the document is not well formed
+     * @throws IOException
+     *             if there are reading issues
+     */
+    private static final Element parseXML(InputStream input, Optional<String> encoding)
+            throws SAXException, IOException {
+        try {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            factory.setNamespaceAware(true);
+            DocumentBuilder documentBuilder;
+            try {
+                documentBuilder = factory.newDocumentBuilder();
+            } catch (ParserConfigurationException e) {
+                String message = e.getMessage();
+                throw new RuntimeException(message != null ? message : e.getClass().getSimpleName(), e);
+            }
+            InputSource source = new InputSource(input);
+            encoding.ifPresent((value) -> {
+                source.setEncoding(value);
+            });
+            return documentBuilder.parse(source).getDocumentElement();
+        } finally {
+            input.close();
+        }
+    }
+
+    /**
+     * Parses an XML DOM tree to a linked data tree.
+     *
+     * @param root
+     *            root node of the XML DOM tree
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @param storage
+     *            storage to read the XML into
+     * @return the root node of a linked data tree
+     */
+    private static Node toNode(Element root, String documentNS, Storage storage) {
+        return parseNodeElement(root, "", Space.DEFAULT, documentNS, true, storage);
+    }
+
+    /**
+     * Creates a Node from a file.
+     *
+     * @param path
+     *            file to read
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the linked data node
+     * @throws SAXException
+     *             if the file is not wellformed
+     * @throws IOException
+     *             if the reading fails
+     */
+    public static Node toNode(File path, Storage storage) throws SAXException, IOException {
```

Comment:
See my comment for accessing file in for example process directory.

Reply:
This is part of reading configuration XML files and rule sets, too.

Reply:
I hope, you are right.